extension/src/views/defaultProject/DefaultProjectProvider.ts

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { Range } from 'vscode-languageclient/node';
+import { GradleClient } from '../../client';
+import { isLanguageServerStarted } from '../../languageServer/languageServer';
+import { RootProject } from '../../rootProject';
+import { GradleTaskDefinition } from '../../tasks';
+import { buildTaskId, createTaskFromDefinition } from '../../tasks/taskUtil';
+import { DependencyConfigurationTreeItem } from '../gradleTasks/DependencyConfigurationTreeItem';
+import { DependencyTreeItem } from '../gradleTasks/DependencyTreeItem';
+import { HintItem } from '../gradleTasks/HintItem';
+import { ProjectDependencyTreeItem } from '../gradleTasks/ProjectDependencyTreeItem';
+
+export class DefaultProjectProvider {
+  // task name, task group, task description
+  private static defaultTaskDefinitions: [string, string, string][] = [
```

Comment:
string[][] is hard to maintain, because statement like `taskDefinition[1]` is hard to understand.

Reply:
Here we can use an interface and an array to refactor, making it more readable.